src/components/widgets/sparkline/sparkline_widget.js

```diff
@@ -0,0 +1,48 @@
+import React, {PureComponent} from 'react';
+import Chart from './../../chart';
+import {makeChartOptions} from './sparkline_dataHelpers';
+import TrendLegend from './../../trendLegend';
+
+/**
+ * Renders a Line Widget with its surrounding state.
+ */
+class SparklineWidget extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.proxiedSetState = this.proxiedSetState.bind(this);
+    this.chartInstance = null;
+    this.state = {
+      customLegend: []
```

Comment:
would be happy to change this to `trendLegend` to be more declarative, and/or happy if stay the same